index.js

```diff
@@ -11,8 +12,15 @@ export default function (folderPath, options = {}) {
   if (!inputs.length) throw `Nothing found at ${realPath} with the suffix .yml or .yaml`
 
   const tokens = inputs.map(processFile).flat(Infinity)
-  const shakenTokens = tokens // TODO: treeshake
-  // TODO: check options.requirements
+  const shakenTokens = options.treeshake
+    ? treeshake(tokens)
+    : tokens
+  try {
+    requireTokens(shakenTokens, options.requirements)
+  } catch (err) {
+    console.error(err, `in directory ${realPath}`)
+    process.exit(1)
+  }
 
   const darkTokens = shakenTokens.filter(t => t.isDark).map(getCSS)
   const normalTokens = shakenTokens.filter(t => !t.isDark).map(getCSS)
```

Comment:
If the list of tokens that are used for generating css can be shorter due to treeshaking, would it make sense to run `toCSSDef` and `toCSSMap` as part of `getCSS`?

Reply:
If this was some kind of realtime app, definitely!

Those functions have almost zero performance impact, and this module doesn't need to be performant - so it's not worth investigating/changing IMO.